waiter/src/waiter/core.clj

```diff
@@ -613,11 +613,13 @@
                        service-id->password-fn*
                        service-id->service-description-fn*]
                 (let [is-waiter-app?-fn (fn is-waiter-app? [^String service-id]
-                                          (str/starts-with? service-id service-id-prefix))]
-                  (utils/create-component scheduler-config :context {:is-waiter-app?-fn is-waiter-app?-fn
-                                                                     :leader?-fn leader?-fn
-                                                                     :service-id->password-fn service-id->password-fn*
-                                                                     :service-id->service-description-fn service-id->service-description-fn*})))
+                                          (str/starts-with? service-id service-id-prefix))
+                      scheduler-context {:is-waiter-app?-fn is-waiter-app?-fn
+                                         :leader?-fn leader?-fn
+                                         :service-id->password-fn service-id->password-fn*
+                                         :service-id->service-description-fn service-id->service-description-fn*}]
+                  (-> (utils/create-component scheduler-config :context scheduler-context)
+                      (scheduler/attach-name (-> scheduler-config :kind name)))))
```

Comment:
I think it would be better to use our `keyword->str` function here in place of `name` so that namespaces are preserved. That way we could name schedulers things like `marathon/production` or `kubernetes/prod/1` or something if we really wanted to.

Reply:
Sounds good. Will change.

Reply:
Done.